src/components/Sliding/index.js

```diff
@@ -0,0 +1,226 @@
+
+  this.selected.get('previous-button').disabled =
+    this.getLoopMode() === 'normal';
+
+  makeSwipable(this.selected.get('sliding-content'));
+
+  initializeSwiper();
+
+  this.selected
+    .get('sliding-content')
+    .addEventListener('swipe-left', () => this.next());
+  this.selected
+    .get('sliding-content')
+    .addEventListener('swipe-right', () => this.prev());
+
+  this.selected
+    .get('previous-button')
+    .addEventListener('click', () => this.prev());
+
+  this.selected.get('next-button').addEventListener('click', () => this.next());
+
+  this.listen('mount', () =>
+    document.addEventListener('click', this.setSlide(this.items[0])),
+  );
+
+  this.listen('unmount', () =>
+    document.addEventListener('click', this.clearItems()),
+  );
+
+  this.getCurrentIndex = () =>
+    this.items.findIndex((item) =>
+      item.classList.contains('sliding__content__slide--selected'),
+    );
+
+  this.addNextAnimation = () => {
+    const slide = this.getSlide();
+
+    slide.classList.remove('sliding__content__slide__animation--prev');
+    slide.classList.add('sliding__content__slide__animation--next');
+  };
+
+  this.addPrevAnimation = () => {
+    const slide = this.getSlide();
+
+    slide.classList.remove('sliding__content__slide__animation--next');
+    slide.classList.add('sliding__content__slide__animation--prev');
+  };
+
+  this.updateStatusControls = () => {
+    const currentIndex = this.getCurrentIndex();
+
+    this.selected.get('previous-button').disabled = currentIndex === 0;
+
+    this.selected.get('next-button').disabled =
+      currentIndex === this.items.length - 1;
+
+    this.emit('controls:disabled');
+  };
+}
+
+Sliding.prototype = Object.assign(Sliding.prototype, Component.prototype, {
+  addItem(item) {
+    item.classList.add('sliding__content__slide');
+    this.selected.get('sliding-content').appendChild(item);
+
+    this.updateItems();
+
+    this.emit('item:add', item);
+  },
+
+  removeItem(item) {
+    const itemExist = this.selected.get('sliding-content').contains(item);
+
+    if (!itemExist) throw new Error('Item not exists');
+
+    this.selected.get('sliding-content').removeChild(item);
+
+    this.updateItems();
+
+    this.emit('item:remove', item);
+  },
+
+  setSlide(item) {
+    this.items.forEach((i) => {
+      if (item === i) {
+        i.classList.add('sliding__content__slide--selected');
+      } else {
+        i.classList.remove('sliding__content__slide--selected');
+      }
+    });
+
+    if (this.getLoopMode() === 'normal') this.updateStatusControls();
+
+    this.emit('slide:change', item);
+  },
+
+  getSlide() {
+    return this.items.find((item) =>
+      item.classList.contains('sliding__content__slide--selected'),
+    );
+  },
+
+  next() {
+    let currentIndex = this.getCurrentIndex();
+
+    if (
+      this.getLoopMode() === 'normal' &&
+      currentIndex === this.items.length - 1
+    )
+      throw new Error('There are no next items');

```

Comment:
nao sei se é legal jogar um erro se nao tiver o proximo item no slider por que isso vai quebrar a aplicação, talvez só um return vazio seja melhor

```suggestion
      return
```